operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java

```diff
@@ -513,6 +513,7 @@ public void makeRightDomainPresence(
           // change just by looking at metadata.generation.
 
           // Has the spec actually changed? We will get watch events for status updates
+          DomainPresenceControl.normalizeDomainSpec(current.getSpec());
```

Comment:
The problem with the call to 'normalize' is that it puts control of the internals of domain matching outside the domain classes. We should really fix this by correcting the equals methods to use the effective values of things rather than the raw fields.

Reply:
Removed DomainPresenceControl class and used effective default values for 'image' and 'imagePullPolicy' directly in DomainSpec class.  DomainSpec.equals() method uses those effective values rather than raw fields by calling getImage() and getImagePullPolicy().